source/configuration_manual/authentication/ldap_conf_ext_settings.rst

```diff
@@ -0,0 +1,603 @@
+Example:
+
+.. code-block:: none
+
+   iterate_filter = (objectClass=smiMessageRecipient)
+
+
+.. _ldap_conf_ext_setting-ldaprc_path:
+
+``ldaprc_path``
+---------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+
+If a non-empty value is set, it will be set to the LDAPRC environment variable.
+
+
+.. _ldap_conf_ext_setting-ldap_version:
+
+``ldap_version``
+----------------
+
+- Default: ``3``
+- Values:  :ref:`uint`
+
+LDAP protocol version to use. Likely 2 or 3.
+
+
+.. _ldap_conf_ext_setting-pass_attrs:
+
+``pass_attrs``
+--------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Specify user attributes to be retrived from LDAP in passdb look up.
+See also :ref:`authentication-ldap_backend_configuration`
+
+Password checking attributes:
+* user: Virtual user name (user@domain), if you wish to change the user-given username to something else
+* password: Password, may optionally start with {type}, eg. {crypt}
+
+Example:
+
+.. code-block:: none
+
+   pass_attrs = uid=user,userPassword=password,\
+                homeDirectory=userdb_home,uidNumber=userdb_uid,gidNumber=userdb_gid
+
+
+There are also other special fields which can be returned.
+See :ref:`authentication-password_database_extra_fields`
+
+If you wish to avoid two LDAP lookups (passdb + userdb), you can use
+userdb prefetch instead of userdb ldap in dovecot.conf. In that case you'll
+also have to include user_attrs in pass_attrs field prefixed with ``userdb_``
+string.
+
+
+.. _ldap_conf_ext_setting-pass_filter:
+
+``pass_filter``
+---------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+Filter for password lookups (passdb lookup)
+See also :ref:`authentication-ldap_backend_configuration`
+
+Example:
+
+.. code-block:: none
+
+   pass_filter = (&(objectClass=posixAccount)(uid=%u))
+
+
+.. _ldap_conf_ext_setting-sasl_authz_id:
+
+``sasl_authz_id``
+-----------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+SASL authorization ID, ie. the dnpass is for this "master user", but the
+dn is still the logged in user. Normally you want to keep this empty.
+
+
+.. _ldap_conf_ext_setting-sasl_bind:
+
+``sasl_bind``
+-------------
+
+- Default: ``no``
+- Values:  :ref:`boolean`
+
+Set yes to use SASL binding instead of the simple binding. Note that this changes
+ldap_version automatically to be 3 if it's lower.
+
+
+.. _ldap_conf_ext_setting-sasl_mech:
+
+``sasl_mech``
+-------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+SASL mechanism names (a space-separated list of candidate mechanisms) to use.
+
+.. todo:: may need to list such mech names?
+
+
+.. _ldap_conf_ext_setting-sasl_realm:
+
+``sasl_realm``
+--------------
+
+- Default: <empty>
+- Values:  :ref:`string`
+
+SASL realm to use.
+
+
+.. _ldap_conf_ext_setting-scope:
+
+``scope``
+---------
+
+- Default: ``subtree``
+- Values:  ``base, onelevel, subtree``
+
+This specifies the search scope.
+
+
+.. _ldap_conf_ext_setting-sieve_ldap_filter:
+
+``sieve_ldap_filter``
+---------------------

```

Comment:
I think we need to split this into 3 pages:
 * Common LDAP settings - need to figure out some place for this..
 * auth LDAP settings
 * sieve LDAP settings

Now it's a bit confusing that auth and sieve settings are mixed together, since they're not available together in the same places.

Reply:
I noticed that there is already a file for ldap sieve specific parameters, configuration_manual/sieve/ldap.rst. ldap_conf_ext is now devided to ldap_settings_common.rst and ldap_settings_auth.rst, and sieve/ldap.rst has a ref to ldap_settings_common.rst .